data-prepper-plugins/opensearch-source/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSourceConfigurationTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.opensearch;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import org.junit.Test;
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.AwsAuthenticationConfiguration;
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.ConnectionConfiguration;
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.SchedulingParameterConfiguration;
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.SearchConfiguration;
+import org.opensearch.dataprepper.plugins.source.opensearch.configuration.WildCardOptions;
+import software.amazon.awssdk.regions.Region;
+
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class OpenSearchSourceConfigurationTest {
+
+    private ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory().enable(YAMLGenerator.Feature.USE_PLATFORM_LINE_BREAKS));
+
+    @Test
+    public void open_search_source_config_values_test() throws JsonProcessingException {
+
+        final String sourceConfigurationYaml = "max_retries: 5\n" +
+                "hosts: [\"http://localhost:9200\"]\n" +
+                "username: test\n" +
+                "password: test\n" +
+                "connection:\n" +
+                "  insecure: true\n" +
+                "  cert: \"cert\"\n" +
+                "indices:\n" +
+                "  include:\n" +
+                "    - \"shakespeare\"\n" +
+                "aws:\n" +
+                "  region: \"us-east-1\"\n" +
+                "  sts_role_arn: \"arn:aws:iam::123456789012:role/aos-role\"\n" +
+                "scheduling:\n" +
+                "  rate: \"P2DT3H4M\"\n" +
```

Comment:
If testing this is a problem without the duration deserializer, just remove it from here since that is an optional setting.